mipengine/algorithms/anova.py

```diff
@@ -5,12 +5,21 @@
 import scipy.stats as st
 from pydantic import BaseModel
 
+from mipengine.algorithm_specification import AlgorithmSpecification
+from mipengine.algorithm_specification import InputDataSpecification
+from mipengine.algorithm_specification import InputDataSpecifications
+from mipengine.algorithm_specification import InputDataStatType
+from mipengine.algorithm_specification import InputDataType
+from mipengine.algorithm_specification import ParameterSpecification
+from mipengine.algorithm_specification import ParameterType
```

Comment:
Maybe, instead of importing a bunch of stuff from `algorithm_specification` we could just import a factory function or class, passing everything as simple python types (dict, lists, str, etc), which would then take care of instantiating the `AlgorithmSpecification` object. The passed dict would basically look a lot like the old `json` file. The factory should be properly documented and type annotated, to help algorithm devs.

Less boilerplate and easier extensibility. For example, if a spec changes only `mipengine.algorithm_specification` changes.

Reply:
I am not sure what you are suggesting. Are you suggesting to just have a dict here just like in the json instead of python objects?

Reply:
As decided in a call we will not go forward with this approach.